templates/csharp/SmartContract1.cs

```diff
@@ -10,8 +12,30 @@
 
 public class SmartContract1 : SmartContract
 {
-    public static void Main()
+    public static object Main(string method, object[] args)
+    {
+        if (method == "PutValue") 
```

Comment:
Typically, contract methods are camelCased (i.e. putValue). Since C# naming convention is PascalCased, contracts often the DisplayName attribute to override the method names in the abi.json file. Example: https://github.com/neo-ngd/CNEO-Contract/blob/14f6adb46951a2264a3b48d8e9647df32ed95662/NeoContract/CNEO.cs#L112

Reply:
Though not always: https://github.com/Moonlight-io/moonlight-ico-template/blob/8a2ddfa86374b93f3fcc3a340c4f1abb4e3f44ee/ICOTemplate/Token/Administration.cs#L52